docs/sources/installation/istio.md

```diff
@@ -0,0 +1,179 @@
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-query-frontend
+spec:
+  ports:
+  - appProtocol: http
+    name: http
+    port: 3100
+    protocol: TCP
+    targetPort: http
+  - appProtocol: tcp
+    name: grpclb
+    port: 9095
+    protocol: TCP
+    targetPort: grpc
+  publishNotReadyAddresses: true
+  selector:
+    app: loki-query-frontend
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-query-frontend
+  type: ClusterIP
+```
+
+### Querier service
+
+Set the `appProtocol` of the `grpc` service to `tcp`
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: loki-querier
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-querier
+  name: loki-querier
+  namespace: observability
+spec:
+  ports:
+  - appProtocol: http
+    name: http
+    port: 3100
+    protocol: TCP
+    targetPort: http
+  - appProtocol: tcp
+    name: grpc
+    port: 9095
+    protocol: TCP
+    targetPort: grpc
+  selector:
+    app: loki-querier
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-querier
+  type: ClusterIP
+
+```
+
+### Ingester Service and Ingester Headless Service
+
+Set the `appProtocol` of the `grpc` port to `tcp` 
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: loki-ingester-(headless)
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-ingester
+  name: loki-ingester-headless
+spec:  
+  clusterIP: None (if headless)
+  ports:
+  - name: http
+    port: 3100
+    protocol: TCP
+    targetPort: http
+  - appProtocol: tcp
+    name: grpc
+    port: 9095
+    protocol: TCP
+    targetPort: grpc
+  selector:
+    app: loki-ingester
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-ingester
+  type: ClusterIP
+```
+
+### Distributor Service
+
+Set the `appProtocol` of the `grpc` port to `tcp` 
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: loki-distributor
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-distributor
+spec:
+  ports:
+  - name: http
+    port: 3100
+    protocol: TCP
+    targetPort: http
+  - name: grpc
+    port: 9095
+    protocol: TCP
+    targetPort: grpc
+    appProtocol: tcp
+  selector:
+    app: loki-distributor
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/name: loki-distributor
+  sessionAffinity: None
+  type: ClusterIP
+
+```
+
+## Memberlist Service
+
+Set the `appProtocol` of the `http` port to `tcp`
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/instance: observability
+  name: loki-memberlist
+  namespace: observability
+spec:
+  clusterIP: None
+  ports:
+    - name: http
+      port: 7946
+      protocol: TCP
+      targetPort: 7946
+      appProtocol: tcp
+  selector:
+    app.kubernetes.io/instance: observability
+    app.kubernetes.io/part-of: memberlist
+```
```

Comment:
Do you need to restart all of the service pods for these configuration changes to take effect?